### PR TITLE
HTML-797 Add Velocity date addition and utility functions

### PR DESCRIPTION
**Pull request by ssmusoke**
## Description of what I changed

Add date Velocity addition and utility functions 

## Issue I worked on

see https://issues.openmrs.org/browse/HTML-797

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

Yes

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

Yes

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

Yes

- [x] All new and existing **tests passed**.

Yes

- [x] My pull request is **based on the latest changes** of the master branch.

Yes

### CODE REVIEW

**Review by mseaton — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/htmlformentry/VelocityFunctions.java [hunk 1]
```diff
@@ -570,4 +567,50 @@ public String convertToClientTimezone(Date date) {
...
+		
+		cal.add(Integer.parseInt(interval), Integer.parseInt(value));
+		
+		return cal.getTime();
+	}
+
+	/**
+	 * Date utility functions useful for HTML forms
+	 */
+	public Date yesterday(String date) {
```
[the remark this answers is not in the document]
Comment (mogoodrich) [reply api/src/main/java/org/openmrs/module/htmlformentry/VelocityFunctions.java:600]: Makes sense to me, thanks @mseaton !

[the remark this answers is not in the document]
Comment (ssmusoke) [reply api/src/main/java/org/openmrs/module/htmlformentry/VelocityFunctions.java:600]: @mseaton Thanks for the feedback let me update the PR - makes alot of sense

[the remark this answers is not in the document]
Comment (ssmusoke) [reply api/src/main/java/org/openmrs/module/htmlformentry/VelocityFunctions.java:625]: @mseaton Requested changes added to the PR, I actually feel much better about it now

File: api/src/main/java/org/openmrs/module/htmlformentry/VelocityFunctions.java [hunk 2]
```diff
@@ -570,4 +567,75 @@ public String convertToClientTimezone(Date date) {
...
+		}
+		catch (ParseException e) {
+			throw new IllegalArgumentException(e);
+		}
+	}
+	
+	public Date addDays(Date date, String value) {
+		Calendar cal = Calendar.getInstance();
+		cal.setTime(date);
+
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/htmlformentry/VelocityFunctions.java:590]: How about this? `org.joda.time.LocalDate.fromDateFields(date).plusDays(Integer.parseInt(value))`

File: api/src/main/java/org/openmrs/module/htmlformentry/VelocityFunctions.java [hunk 3]
```diff
@@ -570,4 +567,75 @@ public String convertToClientTimezone(Date date) {
...
+		catch (ParseException e) {
+			throw new IllegalArgumentException(e);
+		}
+	}
+	
+	public Date addMonths(Date date, String value) {
+		Calendar cal = Calendar.getInstance();
+		cal.setTime(date);
+		
+		cal.add(Calendar.MONTH, Integer.parseInt(value));
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/htmlformentry/VelocityFunctions.java:617]: How about this? `org.joda.time.LocalDate.fromDateFields(date).plusMonths(Integer.parseInt(value))`

File: api/src/main/java/org/openmrs/module/htmlformentry/VelocityFunctions.java (hunk 2 above)
Comment (ssmusoke) [reply api/src/main/java/org/openmrs/module/htmlformentry/VelocityFunctions.java:590]: @dkayiwa Works, I had not looked into the Joda time option now I see its an included dependency & much cleaner

File: api/src/main/java/org/openmrs/module/htmlformentry/VelocityFunctions.java (hunk 3 above)
Comment (ssmusoke) [reply api/src/main/java/org/openmrs/module/htmlformentry/VelocityFunctions.java]: @dkayiwa Thanks update made

**Review by mseaton — APPROVED** [review]
Looks good.  Thanks @ssmusoke !

**Pull request merged**